### ISSUE TITLE
ship usable systemd-journal-gatewayd

### ISSUE THREAD

**Issue by crawford** [coreos/bugs#570]
<a href="https://github.com/jonboulle"><img src="https://avatars.githubusercontent.com/u/2601015?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [jonboulle](https://github.com/jonboulle)**
_Tuesday Apr 29, 2014 at 22:09 GMT_
_Originally opened as https://github.com/coreos/coreos-overlay/issues/569_

----

We ship `systemd-journal-gatewayd` (with enabled `service`/`socket` files) but it doesn't work out of the box because it's missing its default user:

```
ip-172-31-14-67 ~ # curl localhost:19531
curl: (56) Recv failure: Connection reset by peer
ip-172-31-14-67 ~ # journalctl -f -u systemd-journal-gatewayd
-- Logs begin at Wed 2014-04-16 18:58:43 UTC. --
Apr 29 22:02:52 ip-172-31-14-67 systemd[16159]: Failed at step USER spawning /usr/lib/systemd/systemd-journal-gatewayd: No such process
Apr 29 22:02:52 ip-172-31-14-67 systemd[1]: systemd-journal-gatewayd.service: main process exited, code=exited, status=217/USER
Apr 29 22:02:52 ip-172-31-14-67 systemd[1]: Unit systemd-journal-gatewayd.service entered failed state.
Apr 29 22:02:52 ip-172-31-14-67 systemd[1]: Starting Journal Gateway Service...
Apr 29 22:02:52 ip-172-31-14-67 systemd[1]: Started Journal Gateway Service.
Apr 29 22:02:52 ip-172-31-14-67 systemd[1]: systemd-journal-gatewayd.service: main process exited, code=exited, status=217/USER
Apr 29 22:02:52 ip-172-31-14-67 systemd[1]: Unit systemd-journal-gatewayd.service entered failed state.
Apr 29 22:02:52 ip-172-31-14-67 systemd[1]: Starting Journal Gateway Service...
Apr 29 22:02:52 ip-172-31-14-67 systemd[1]: systemd-journal-gatewayd.service start request repeated too quickly, refusing to start.
Apr 29 22:02:52 ip-172-31-14-67 systemd[1]: Failed to start Journal Gateway Service.
ip-172-31-14-67 ~ # getent passwd systemd-journal-gatewayd
ip-172-31-14-67 ~ # 
```

(incidentally, [this](https://bugs.freedesktop.org/show_bug.cgi?id=67958) is lame)

I'm totally newb at cros but my guess would be this is because we use a hard-coded passwd instead of letting the ebuild [do its thing](https://github.com/coreos/coreos-overlay/blob/master/sys-apps/systemd/systemd-212-r4.ebuild#L449)?

Can we fix this? @marineam I'd be happy to submit a patch if you could advise how you'd prefer it was done.

**Comment by crawford** [comment]
<a href="https://github.com/jonboulle"><img src="https://avatars.githubusercontent.com/u/2601015?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jonboulle](https://github.com/jonboulle)**
_Tuesday Apr 29, 2014 at 22:10 GMT_

----

From my totally unscientific experiment it doesn't look like it's a problem with anything else we ship out of the box:
```
ip-172-31-14-67 ~ # systemctl list-unit-files --no-legend --no-pager  | awk '{print $1}' | xargs systemctl cat --no-pager 2>/dev/null --  | fgrep User=
User=etcd
User=systemd-journal-gateway
ip-172-31-14-67 ~ # 
```

**Comment by crawford** [comment]
<a href="https://github.com/marineam"><img src="https://avatars.githubusercontent.com/u/153640?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [marineam](https://github.com/marineam)**
_Tuesday Apr 29, 2014 at 22:21 GMT_

----

The ebuild rule doesn't work because the assorted e* user commands only work when installing ebuilds to ROOT=/. So it can install users in the chrooted SDK but not in the in the board build itself because ROOT=/build/amd64-usr and whatnot. Even if it did work we wouldn't want to depend on its behavior because most of those calls in ebuilds do not specify the UID/GID, letting the system pick something that happens to be available. For board builds we have to ensure that UID/GIDs never change between independent builds.

The passwd/group files are managed here: https://github.com/coreos/baselayout/tree/master/baselayout

**Comment by crawford** [comment]
<a href="https://github.com/jonboulle"><img src="https://avatars.githubusercontent.com/u/2601015?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jonboulle](https://github.com/jonboulle)**
_Tuesday Apr 29, 2014 at 22:55 GMT_

----

coreos/baselayout#15

**Comment by crawford** [comment]
<a href="https://github.com/jonboulle"><img src="https://avatars.githubusercontent.com/u/2601015?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jonboulle](https://github.com/jonboulle)**
_Wednesday Apr 30, 2014 at 17:21 GMT_

----

Fixed by coreos/baselayout#15
